src/components/Tabs/Tabs.tsx

```diff
@@ -14,26 +14,37 @@ interface ITabsProps {
 
 export const Tabs = ({ children }: ITabsProps) => {
   const [showScrollButton, setShowScrollButton] = useState(false);
-  const tabContent = document.getElementById('tab-content');
+  const [disableScrollLeft, setDisableScrollLeft] = useState(true);
+  const [disableScrollRight, setDisableScrollRight] = useState(false);
+  const tabContent = useRef<HTMLUListElement>(null);
 
   const onResize = useCallback(() => {
-    tabContent && setShowScrollButton(tabContent.scrollWidth > window.innerWidth);
+    tabContent.current && setShowScrollButton(tabContent.current.scrollWidth > window.innerWidth);
   }, [tabContent]);
 
-  window.addEventListener('resize', onResize);
-
   const scrollLeft = () => {
-    if (tabContent) {
-      tabContent.scrollLeft = tabContent.scrollLeft ? tabContent.scrollLeft - 150 : 0;
+    if (tabContent.current) {
+      setDisableScrollRight(false);
+      setDisableScrollLeft(tabContent.current.scrollLeft - 150 <= 0);
+      tabContent.current.scrollLeft = tabContent.current.scrollLeft ? tabContent.current.scrollLeft - 150 : 0;
     }
   };
+
   const scrollRight = () => {
-    if (tabContent) {
-      tabContent.scrollLeft = tabContent.scrollLeft ? tabContent.scrollLeft + 150 : 150;
+    if (tabContent.current) {
+      setDisableScrollLeft(false);
+      setDisableScrollRight(tabContent.current.scrollWidth <= window.innerWidth + tabContent.current.scrollLeft + 150);
```

Comment:
`scrollLeft` / `scrollRight` callbacks are run only when respective buttons are pressed.
This does not take into mind the usage of the touchpad / mouse scroll button.
Therefore, these two set functions need to be moved to some scroll event listener; solution below should work.

```
<ul
  ref={tabContent}
  onScroll={() => {
    if (tabContent.current) {
      setDisableScrollLeft(tabContent.current.scrollLeft - 5 <= 0);
      setDisableScrollRight(
        tabContent.current.scrollLeft + tabContent.current.offsetWidth + 5 >= tabContent.current.scrollWidth
      );
    }
  }}
  className={stylesPF.tabsList}
>
  {children}
</ul>
```

+/- 5 just due to he inability to scroll to the far right/left on certain screen sizes.

Reply:
Got it, thanks for raising up the solution!